test/Microsoft.ML.Tests/CollectionDataSourceTests.cs

```diff
@@ -0,0 +1,209 @@
+
+                Assert.NotNull(output.Data);
+                Assert.NotNull(output.Data.VarName);
+                Assert.Null(output.Model);
+            }
+        }
+
+        [Fact]
+        public void CanSuccessfullyEnumerated()
+        {
+            var collection = CollectionDataSource.Create(new List<Input>() {
+                new Input { Number1 = 1, String1 = "1" },
+                new Input { Number1 = 2, String1 = "2" },
+                new Input { Number1 = 3, String1 = "3" }
+            });
+
+            using (var environment = new TlcEnvironment())
+            {
+                Experiment experiment = environment.CreateExperiment();
+                ILearningPipelineDataStep output = collection.ApplyStep(null, experiment) as ILearningPipelineDataStep;
+
+                experiment.Compile();
+                collection.SetInput(environment, experiment);
+                experiment.Run();
+
+                IDataView data = experiment.GetOutput(output.Data);
+                Assert.NotNull(data);
+
+                using (var cursor = data.GetRowCursor((a => true)))
+                {
+                    var IDGetter = cursor.GetGetter<float>(0);
+                    var TextGetter = cursor.GetGetter<DvText>(1);
+
+                    Assert.True(cursor.MoveNext());
+
+                    float ID = 0;
+                    IDGetter(ref ID);
+                    Assert.Equal(1, ID);
+
+                    DvText Text = new DvText();
+                    TextGetter(ref Text);
+                    Assert.Equal("1", Text.ToString());
+
+                    Assert.True(cursor.MoveNext());
+
+                    ID = 0;
+                    IDGetter(ref ID);
+                    Assert.Equal(2, ID);
+
+                    Text = new DvText();
+                    TextGetter(ref Text);
+                    Assert.Equal("2", Text.ToString());
+
+                    Assert.True(cursor.MoveNext());
+
+                    ID = 0;
+                    IDGetter(ref ID);
+                    Assert.Equal(3, ID);
+
+                    Text = new DvText();
+                    TextGetter(ref Text);
+                    Assert.Equal("3", Text.ToString());
+
+                    Assert.False(cursor.MoveNext());
+                }
+            }
+        }
+
+        [Fact]
+        public void CanTrain()
+        {
+            var pipeline = new LearningPipeline();
+            var data = new List<IrisData>() {
+                new IrisData { SepalLength = 1f, SepalWidth = 1f ,PetalLength=0.3f, PetalWidth=5.1f, Label=1},
+                new IrisData { SepalLength = 1f, SepalWidth = 1f ,PetalLength=0.3f, PetalWidth=5.1f, Label=1},
+                new IrisData { SepalLength = 1.2f, SepalWidth = 0.5f ,PetalLength=0.3f, PetalWidth=5.1f, Label=0}
+            };
+            var collection = CollectionDataSource.Create(data);
```

Comment:
Would we address the inconsistency in naming/usage between CollectionDataSource and TextLoader in a separate PR? We want it to be clearer these are the two (current) approaches to bringing data into a LearningPipeline.